src/ripple/basics/impl/contract.cpp

```diff
@@ -39,6 +41,18 @@ accessViolation() noexcept
 
 } // detail
 
+void
+LogThrow (std::string title)
+{
+    if (auto j = debugJournal().debug())
+    {
+        j << title;
+        for (auto const& frame : beast::getStackBacktrace ())
+            j << "    " << frame;
+        j << "----";
+    }
+}
```

Comment:
I've already discussed this with Nik, but I'm repeating here to get the word out to a wider audience:  On OS X we don't want to catch, log and rethrow at the top of the thread as this destroys information on OS X.  We get a better stack dump without the catch/rethrow.  Unfortunately gcc operates differently, with a lousy stack dump whatever we do.  `#ifndef __APPLE__` can be used to guard such changes.